### PR TITLE
Issue979

### LINKED ISSUE #979: Magma protocol losing type under ite
Bit.ite does not "cast" protocols back to their protocol type

Specifically:
```Python
assert type(proto0) == type(proto1)
v = Bit().ite(proto0, proto1)
assert type(v) == type(proto0)._to_magma_
```
where I would expect:
```Python
assert type(proto0) == type(proto1)
v = Bit().ite(proto0, proto1)
assert type(v) == type(proto0)
```

### PR DESCRIPTION
**Pull request by cdonovick**
closes #979

### CODE REVIEW

File: magma/bit.py [hunk 1]
```diff
@@ -161,7 +161,13 @@ def ite(self, t_branch, f_branch):
         # Note: coreir flips t/f cases
         # self._mux monkey patched in magma/primitives/mux.py to avoid circular
         # dependency
-        return self._mux([f_branch, t_branch], self)
+        r = self._mux([f_branch, t_branch], self)
+
+        # cast protocol values back to protocol
```
Comment (rsetaluri) [inline magma/bit.py:166]: wondering if we want to add the conversion function in `Mux` generator (https://github.com/phanrahan/magma/blob/0975d81d192918dcee16194e03b4f1912c547841/magma/primitives/mux.py#L38) or `mux` helper function (https://github.com/phanrahan/magma/blob/0975d81d192918dcee16194e03b4f1912c547841/magma/primitives/mux.py#L126).

@leonardt

Comment (cdonovick) [reply magma/bit.py:166]: doesn't matter to me I was just was looking the fastest way to get a working solution

Comment (rsetaluri) [reply magma/bit.py:166]: ok give me a little bit, i'll fix the mux and it should fix your issue

Comment (cdonovick) [reply magma/bit.py:166]: Thanks, sounds good to me.  As long as it passes my test I don't care where the casting logic is.

**Review by leonardt — APPROVED** [review]
LGTM, moving it to the mux seems reasonable and should provided consistent behavior when using the mux operator directly

**Pull request merged**